BrowserKit/Sources/Common/Theming/DefaultThemeManager.swift

```diff
@@ -216,10 +217,13 @@ public final class DefaultThemeManager: ThemeManager, Notifiable {
     }
 
     private func updateSavedTheme(to newTheme: ThemeType) {
-        // We never want to save the private theme because it's meant to override
-        // whatever current theme is set. This means that we need to know the theme
-        // before we went into private mode, in order to be able to return to it.
-        guard newTheme != .privateMode else { return }
+        // We never want to save the private or night mode theme because it's meant
+        // to override whatever current theme is set. This means that we need to
+        // know the theme before we went into private mode, in order to be able
+        // to return to it.
+        guard newTheme != .privateMode,
+              newTheme != .nightMode
```

Comment:
Just a thought but I wonder if this would make sense to add as an extension on `ThemeType`, something like:

```
extension ThemeType {
     var isOverrideThemeType {
         switch self {
             case .nightMode, .privateMode: return true
             case .dark, .light: return false
         }
     }
}
```

And then checking that here:

`guard !newTheme.isOverrideThemeType else { return }`

The advantage would be that new theme types would always have to update that switch statement to ensure the logic is re-evaluated any time something changes.

Reply:
THat's a great point!